indico/modules/rb_new/client/js/modules/bookRoom/BookRoom.jsx

```diff
@@ -157,6 +158,47 @@ class BookRoom extends React.Component {
         openBookingForm(roomId, {...bookingData, isPrebooking});
     }
 
+    transformToLabel(type) {
+        switch (type) {
```

Comment:
By the way, since the translation is always the same anyway, why not split these exhaustive switches in two declarative structures?
 * An object that maps `style -> translation`
 * A list of style IDs. e.g. `['available', 'booking', 'pre-booking', ...]` (no need for `order`, since the list is already ordered anyway)

Each widget would need to have its own version of the second, but they could all share a common dictionary of translations.